### PR TITLE
feat: [v2]Upgrade to preact X

### PR DESCRIPTION
**Pull request by dongsik-yoo**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

Be adapted about more strict types of preact X.
Change from legacy to new context api. Remove legacy context and related HOC.
Remove preact-context-provider because of above context changes.

All components which refer own instance context have to inherit `ContextComponent` class.
```js
import { Component } from 'preact';
import { InstanceContext } from '@src/model/context';
import { AppContext } from '@src/model';

export default abstract class ContextComponent<P = {}, S = {}> extends Component<P, S> {
  static contextType = InstanceContext;

  context!: AppContext;
}
```
The instance context is widely used across parent and child components.
```js
export interface AppContext {
  options: Option;
  dataStore: DataStore;
  theme: Theme;
  templates: Template;
  internalEvent: EventHandler<InternalEventName>;
  externalEvent: EventHandler<ExternalEventName>;
}
```

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/ts/components/template.tsx [hunk 1]
```diff
@@ -2,23 +2,28 @@ import { h } from 'preact';
...
-  return isString(htmlOrVnode) ? (
-    <Markup wrap={false} markup={htmlOrVnode} allow-events />
-  ) : (
-    htmlOrVnode
-  );
+    return isString(htmlOrVnode) ? (
+      <Markup wrap={false} markup={htmlOrVnode} allow-events />
+    ) : (
+      htmlOrVnode
+    );
```
Comment (seonim-ryu) [inline src/ts/components/template.tsx]: 괄호 없어도 될 것 같습니다.

File: stories/Button.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { h, RenderableProps, ComponentChild } from 'preact';
+import { h, RenderableProps, JSX } from 'preact';
 
-const Button = ({ children, ...props }: RenderableProps<ComponentChild>): JSX.Element => (
+const Button = ({ children, ...props }: RenderableProps<any>): JSX.Element => (
```
Comment (seonim-ryu) [inline stories/Button.tsx]: 이 버튼은 어디서 사용되는건가요??

Comment (dongsik-yoo) [reply stories/Button.tsx]: 스토리북 생성 시 기본 테스트 버튼입니다. 스토리북으로 개발 시작하면 제거될거예요.

File: src/ts/components/template.tsx (hunk 1 above)
Comment (dongsik-yoo) [reply src/ts/components/template.tsx]: 삼항연산자 쓰면 prettier가 괄호 붙여 줍니다.

**Comment by dongsik-yoo** [comment]
@seonim-ryu bower 는 뺄거구요. preact 디펜던시는 포함해서 번들할거기 때문에 상관없어요.

File: src/ts/components/text.tsx [hunk 3]
```diff
@@ -30,8 +31,8 @@ export class Text extends ContextComponent<Props, State> {
     });
```
[the remark this answers is not in the document]
Comment (dongsik-yoo) [reply src/ts/components/text.tsx]: Yes, got it. But the `Text` component is a sample source. It'll be removed.

**Pull request merged**